ion-hash/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+//! Implement Ion Hash as per the [spec](https://amzn.github.io/ion-hash/docs/spec.html)
+//!
+//! ## Examples
+//! ```rust
+//! use ion_rs::value::loader::{self, Loader};
+//! use ion_rs::result::IonResult;
+//!
+//! # fn main() -> IonResult<()> {
+//!   let loader = loader::loader();
+//!   let elem = loader.iterate_over(b"\"hello world\"")?.next().unwrap()?;
+//!   let digest = ion_hash::sha256(&elem);
+//!   println!("{:?}", digest);
+//! # Ok(())
+//! # }
+//! ```
+
+use ion_rs::{
+    value::{owned::OwnedElement, Element},
+    IonType,
+};
+
+use sha2::{digest::Output, Digest, Sha256};
+
+pub fn sha256(elem: &OwnedElement) -> Vec<u8> {
+    let mut hasher = IonHasher::new(Sha256::new());
+    let result = hasher.visit_element(elem);
+    Vec::from(result.as_slice())
+}
+
+/// Bytes markers as per the spec.
+mod markers {
+    /// single byte begin marker
+    pub(crate) const B: u8 = 0x0B;
+    /// single byte end marker
+    pub(crate) const E: u8 = 0x0E;
+    /// single byte escape
+    pub(crate) const ESC: u8 = 0x0C;
+    /// type qualifier octet consisting of a four-bit type code T followed by a four-bit qualifier Q
+    /// (this varies per type)
+    pub(crate) const TQ_STRING: u8 = 0x80;
+}
```

Comment:
Thoughts on associated constants for this--it would eliminate the `pub(crate)` requirement (at least for how it is being used now):

```rust
/// Byte markers as per spec.
struct Marker;

impl Marker {
    /// single byte begin marker
    const B: u8 = 0x0B;
    /// single byte end marker
    const E: u8 = 0x0E;
    /// single byte escape
    const ESC: u8 = 0x0C;
    /// type qualifier octet consisting of a four-bit type code T followed by a four-bit qualifier Q
    /// (this varies per type)
    const TQ_STRING: u8 = 0x80;
}
```

Reply:
Good idea.